src/core/execute.c

```diff
@@ -5147,6 +5148,19 @@ int exec_spawn(Unit *unit,
                 }
         }
 
+        if (context->cgroup_nft_family > 0) {
+                r = nft_set_element_add_uint32(context->cgroup_nft_family,
+                                               context->cgroup_nft_table,
+                                               context->cgroup_nft_set,
+                                               unit->cgroup_id);
```

Comment:
this should be done at cgroup realizatio time, not when we exec a process into it.